azure/src/wizard/RoleAssignmentExecuteStep.ts

```diff
@@ -3,42 +3,50 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { type RoleDefinition } from '@azure/arm-authorization';
 import { AzureWizardExecuteStep, nonNullValueAndProp } from '@microsoft/vscode-azext-utils';
 import { randomUUID } from 'crypto';
 import { l10n, Progress } from 'vscode';
 import * as types from '../../index';
 import { createAuthorizationManagementClient } from '../clients';
 import { ext } from '../extensionVariables';
 
+export interface Role {
+    scopeId: string | undefined;
+    roleDefinitionId: string;
+    roleDefinitionName: string;
+}
+
 export class RoleAssignmentExecuteStep<T extends types.IResourceGroupWizardContext> extends AzureWizardExecuteStep<T> {
     public priority: number = 900;
-    private getScopeId: () => string | undefined;
-    private _roleDefinition: RoleDefinition;
-    public constructor(getScopeId: () => string | undefined, roleDefinition: RoleDefinition) {
+    private roles: () => Role[] | undefined;
+    public constructor(roles: () => Role[] | undefined) {
         super();
-        this.getScopeId = getScopeId;
-        this._roleDefinition = roleDefinition;
+        this.roles = roles;
     }
 
     public async execute(wizardContext: T, progress: Progress<{ message?: string; increment?: number }>): Promise<void> {
         const amClient = await createAuthorizationManagementClient(wizardContext)
-        const scope = this.getScopeId();
-        if (!scope) {
-            throw new Error(l10n.t('No scope was provided for the role assignment.'));
-        }
-        const scopeSplitArr = scope.split('/');
-        const resourceName = scopeSplitArr[scopeSplitArr.length - 1] ?? '';
-        const resourceType = scopeSplitArr[scopeSplitArr.length - 2] ?? '';
+        const roles = this.roles();
+        if (roles) {
+            for (const role of roles) {
```

Comment:
I don't think we would want to make the whole step fail if one role fails. Maybe do a try/catch so it can continue?